pkg/serverhandler/handlertest/test_service_deploy.go

```diff
@@ -0,0 +1,201 @@
+package handlertest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint/pkg/server"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	serverptypes "github.com/hashicorp/waypoint/pkg/server/ptypes"
+)
+
+func init() {
+	tests["deploy"] = []testFunc{
+		TestServiceDeployment,
+		TestServiceDeployment_GetDeployment,
+		TestServiceDeployment_ListDeployments,
+	}
+}
+
+func TestServiceDeployment(t *testing.T, factory Factory) {
```

Comment:
These aren't new - they're from service_deploy_test.go. There was _one_ test over there that violated the interfaces, so I left it over there and brought the tests I could over here.